src/new_index/mempool.rs

```diff
@@ -520,12 +521,24 @@ impl Mempool {
             if new_txids.is_empty() {
                 break;
             }
+
+            let all_txs_count = all_txids.len();
+            let fetched_txs_count = indexed_txids.len() + fetched_txs.len();
+            let new_txs_count = new_txids.len();
+
             debug!(
                 "mempool with total {} txs, {} fetched, {} missing",
-                all_txids.len(),
-                indexed_txids.len() + fetched_txs.len(),
-                new_txids.len()
+                all_txs_count,
+                fetched_txs_count,
+                new_txs_count
             );
+
+            let counter = &mempool.write().unwrap().count;
+
+            counter.with_label_values(&["txs"]).set(all_txs_count as f64);
```

Comment:
The "txs" stat was previously set to the number of transactions in our local mempool view, after we completed processing them.

With this change it will be set to the number of transactions in the bitcoind mempool instead. Was that intentional?

Reply:
no, not intentional. we want to see stats about our local mempool view.
I'll refactor.

Reply:
Maybe log both tx counts, so they could be compared?

Reply:
I refactored, still some duplicated code with brief locks.

Reply:
Also, I think this can take a read lock instead of write.

Reply:
changed to `read` in an amended commit